Add form-submitting mobile worker option to Trigger Invoice testing tools

nospame opened the pull request:
## Product Description
For accounting admin users with the accounting testing tools feature flag enabled, adds a new field to overwrite the number of "form-submitting mobile workers". Also updates help text to reflect this change.
<img width="799" alt="Untitled" src="https://github.com/user-attachments/assets/e833832e-0bc4-4c08-b781-80a6aaac1617">


## Technical Summary
[SAAS-15710](https://dimagi.atlassian.net/browse/SAAS-15710)
Allows overwriting the `FormSubmittingMobileWorkerHistory` for a given domain when testing via Trigger Invoice, which will be useful for QA of changes made in https://github.com/dimagi/commcare-hq/pull/34868. Does not rely directly on changes made in that PR, but does not make sense to add without it.

## Feature Flag
ACCOUNTING_TESTING_TOOLS

## Safety Assurance

### Safety story
This change is limited in access to those with accounting admin access and the user-level feature flag enabled. The feature as it exists has the potential for real data loss if used incorrectly (specifically, overwriting the count of users for a domain from a particular date); this change neither increases nor mitigates that risk.

### Automated test coverage
Added automated testing at `corehq.apps.accounting.tests.test_forms.TestTriggerInvoiceForm` because there didn't seem to be any for this form.

### QA Plan
This change will be used in the QA of #34868 

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change


[SAAS-15710]: https://dimagi.atlassian.net/browse/SAAS-15710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

nospame committed:
Add form-submitting mobile worker option to Trigger Invoice tool

nospame committed:
autosort imports

nospame committed:
Automated testing for TriggerInvoiceForm

biyeun reviewed (commented):
left some comments about the rationale for presenting the history type in this form

biyeun commented:
offlined with Evan. Seemed best to add two number fields, one for num of users and one for num of form-submitting users so that both types of histories can in be generated on submit. Also, a note to take care that errors when generating invoices provide proper guidance in this form if the error is due to one (or both) of the histories being missing

nospame committed:
Use IntegerField to set form-submitting workers

Allows overwriting DomainUserHistory, FormSubmittingMobileWorkerHistory, both, or neither when testing with Trigger Invoice

nospame committed:
Render testing options in separate fieldset for visual distinction

nospame committed:
Merge branch 'master' into em/form-submitting-trigger-invoice
